services/cd-service/pkg/repository/transformer.go

```diff
@@ -708,6 +708,13 @@ func removeCommit(fs billy.Filesystem, commitID, application string) error {
 	}
 
 	commitApplicationDir := commitApplicationDirectory(fs, commitID, application)
+	_, err := fs.ReadDir(commitApplicationDir)
+	if err != nil {
+		// could not read the directory commitApplicationDir - but that's ok, because we don't know
+		// if the kuberpult version that accepted this commit in the release endpoint, did already have commit writing enabled.
+		// So there's no guarantee that this file ever existed
+		return nil
+	}
 	if err := fs.Remove(commitApplicationDir); err != nil {
```

Comment:
Wouldn't it be enough to check if the error here is `fs.ErrNotExist`?

Reply:
done